src/drawing/fill.rs

```diff
@@ -0,0 +1,62 @@
+use image::{GenericImage, Pixel};
+
+/// Equivalent to bucket tool in MS-PAINT
+/// Performs 4-way flood-fill based on this algorithm: https://en.wikipedia.org/wiki/Flood_fill#Span_filling
+pub fn flood_fill<I>(image: &mut I, x: u32, y: u32, fill_with: I::Pixel)
+where
+    I: GenericImage,
```

Comment:
We tend to prefer using the `Image` type rather than `GenericImage` for performance reasons. Also methods that take a `&mut` image should normally end with `_mut`, see other similar methods for examples on creating both variants and using the doc macro. I'm imagining something like this
```suggestion
pub fn flood_fill_mut<P>(image: &mut Image<P>, x: u32, y: u32, fill_with: P)
where
    P: Pixel,
```

Reply:
Done